apps/web/src/components/Community/CommunityGalleryListItem.tsx

```diff
@@ -0,0 +1,182 @@
+import { CommunityGalleryListItemQueryFragment$key } from '~/generated/CommunityGalleryListItemQueryFragment.graphql';
+import { useIsMobileWindowWidth } from '~/hooks/useWindowSize';
+import { EditPencilIcon } from '~/icons/EditPencilIcon';
+
+import GalleryLink from '../core/GalleryLink/GalleryLink';
+import IconContainer from '../core/IconContainer';
+import { HStack, VStack } from '../core/Spacer/Stack';
+import { BaseM } from '../core/Text/Text';
+import { TokenPreview, TokenPreviewContainer } from '../Explore/ExploreUserCard';
+import { ProfilePicture } from '../ProfilePicture/ProfilePicture';
+
+type Props = {
+  communityGalleryRef: CommunityGalleryListItemFragment$key;
+  queryRef: CommunityGalleryListItemQueryFragment$key;
+};
+
+export function CommunityGalleryListItem({ communityGalleryRef, queryRef }: Props) {
+  const communityGallery = useFragment(
+    graphql`
+      fragment CommunityGalleryListItemFragment on CommunityGallery {
+        __typename
+        gallery {
+          dbid
+          name
+          owner {
+            dbid
+            username
+            ...ProfilePictureFragment
+          }
+        }
+        tokenPreviews {
+          large
+        }
+      }
+    `,
+    communityGalleryRef
+  );
+
+  const query = useFragment(
+    graphql`
+      fragment CommunityGalleryListItemQueryFragment on Query {
+        viewer {
+          ... on Viewer {
+            user {
+              dbid
+            }
+          }
+        }
+      }
+    `,
+    queryRef
+  );
+
+  const { gallery } = communityGallery || {};
+
+  const [isHovered, setIsHovered] = useState(false);
+  const isMobile = useIsMobileWindowWidth();
+
+  const isOwnerGallery = useMemo(() => {
+    return gallery?.owner?.dbid === query?.viewer?.user?.dbid;
+  }, [gallery?.owner?.dbid, query?.viewer?.user?.dbid]);
+
+  const tokenPreviews = useMemo(() => {
+    const previews = removeNullValues(communityGallery?.tokenPreviews);
+    const randomIndex = Math.floor(Math.random() * previews.length);
+
+    if (isHovered && previews.length > 2) {
+      return [previews[randomIndex], previews[randomIndex + 1]];
+    }
+    return previews.slice(0, 2);
+  }, [communityGallery?.tokenPreviews, isHovered]);
+
+  const router = useRouter();
+  const handleEditGallery = useCallback(
+    (e: React.MouseEvent<HTMLElement>) => {
+      e.preventDefault();
+      const galleryId = gallery?.dbid;
+
+      if (!galleryId) return;
+      router.push({
+        pathname: '/gallery/[galleryId]/edit',
+        query: { galleryId },
+      });
+    },
+    [gallery?.dbid, router]
+  );
+
+  return (
+    <StyledCommunityGalleryListItem
+      to={{
+        pathname: '/[username]/galleries/[galleryId]',
+        query: {
+          username: communityGallery?.gallery?.owner?.username || '',
+          galleryId: communityGallery?.gallery?.dbid || '',
+        },
+      }}
+      eventElementId="Collection Gallery Card"
```

Comment:
can we rename to "Collection Page Gallery Tab Card" and same for the event name (+"Click"), that'll be helpful when distinguishing between event types 🙏